src/main/scala/main.scala

```diff
@@ -1,4 +1,78 @@
+import scala.io.Source
+import scala.util.{Try, Using}
+import scala.util.matching.Regex
+
 @main
 def main(): Unit = {
-  println("Hello world!")
+  val studentsTry: Try[Seq[(String, Int)]] = Using(Source.fromFile("marks.txt")) { file =>
+    val regexp: Regex = "(\\w+):(\\d+)".r
+
+    def lineToStudent(line: String): Option[(String, Int)] =
+      line match
+        case regexp(name, markStr) => for {
+          mark <- markStr.toIntOption
+        } yield (name, mark)
+        case _ => None
+
+    file.getLines().map(lineToStudent).collect {
+      case Some(value) => value
+    }.toSeq
+  }
+
+  def sumMarks(students: Seq[(String, Int)]): Map[String, Int] =
+    students.groupMapReduce(_._1)(_._2)(_ + _)
+
+  def sortSumMarks(students: Seq[(String, Int)]): Seq[Int] =
+    sumMarks(students).values.toSeq.sorted

```

Comment:
Порядок нужен по убыванию, а не по возрастанию.
```suggestion
    sumMarks(students).values.toSeq.sorted.reverse
```
Интересная интерпретация задания, все остальные выводили отсортированный по баллам список пар (фамилия, сумма баллов)